File: 구름 알고리즘 미션/그래프 탐색(1)/작은노드/작은노드.py
import sys
sys.setrecursionlimit(10**6)
input = sys.stdin.readline

N, M, K = map(int, input().split())
graph = [[] for _ in range(N + 1)]

for _ in range(M):
    s, e = map(int, input().split())
    graph[s].append(e)
    graph[e].append(s)

for i in range(1, N + 1):
    graph[i].sort()

visited = [False] * (N + 1)
count = 0
last_node = 0

def dfs(node):
    global count, last_node
    visited[node] = True
    count += 1

    for next_node in graph[node]:
        if not visited[next_node]:
            dfs(next_node)
            return 

    last_node = node

dfs(K)
print(count, last_node)

File: 구름 알고리즘 미션/그래프 탐색(1)/작은노드/작은노드.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [N, M, K] = input[0].split(" ").map(Number);
const graph = Array.from({length: N+1}, () => []);
for(let i = 1 ; i <= M ; i++){
	const [s, e] = input[i].split(" ").map(Number);
	graph[s].push(e);
	graph[e].push(s);
}

for(let i = 1 ; i <= N ; i++){
	graph[i].sort((a, b) => a - b);
}

const visited = Array(N+1).fill(false);
let cnt = 0;
let lastNode = 0;

const dfs = (nodeNum) => {
	visited[nodeNum] = true;
	cnt++;
	
	for(const node of graph[nodeNum]){
		if(!visited[node]){
			dfs(node);
			return;
		}
	}
	lastNode = nodeNum;
}

dfs(K);
console.log(cnt, lastNode);